testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -10052,3 +10053,408 @@ fn no_reorg_due_to_successive_block_validation_ok() {
+    wait_for(30, || {
+        signer_test.send_status_request(&HashSet::new());
+        thread::sleep(Duration::from_secs(1));
+        let latest_states = signer_test.get_states(&finished_signers);
+        for (ix, state) in latest_states.iter().enumerate() {
+            let Some(state) = state else {
+                continue;
+            };
+            if state.runloop_state == State::RegisteredSigners && state.running_signers.len() == 2 {
+                finished_signers.insert(ix);
+            } else {
+                warn!(
+                    "Signer #{ix} returned state = {:?}, running signers = {:?}. Will try again",
+                    state.runloop_state, state.running_signers
+                );
+            }
+        }
+        Ok(finished_signers.len() == num_signers)
+    })
+    .expect("Timed out while waiting for the signers to be registered for both reward cycles");
+
+    let current_burnchain_height = signer_test
+        .running_nodes
+        .btc_regtest_controller
+        .get_headers_height();
+    assert_eq!(current_burnchain_height, middle_of_prepare_phase);
+    assert_eq!(curr_reward_cycle, signer_test.get_current_reward_cycle());
+
+    let mined_blocks = signer_test.running_nodes.nakamoto_blocks_mined.clone();
+    let blocks_before = mined_blocks.load(Ordering::SeqCst);
+
+    info!("------------------------- Test Mine A Valid Block -------------------------");
+    // submit a tx so that the miner will mine an extra block
+    let sender_nonce = 0;
+    let transfer_tx = make_stacks_transfer(
+        &sender_sk,
+        sender_nonce,
+        send_fee,
+        signer_test.running_nodes.conf.burnchain.chain_id,
+        &recipient,
+        send_amt,
+    );
+    submit_tx(&http_origin, &transfer_tx);
+
+    // a tenure has begun, so wait until we mine a block
+    wait_for(30, || {
+        Ok(mined_blocks.load(Ordering::SeqCst) > blocks_before)
+    })
+    .expect("Timed out waiting for a block to be mined");
+
+    let blocks_before = mined_blocks.load(Ordering::SeqCst);
+    let mut stackerdb = StackerDB::new(
+        &signer_test.running_nodes.conf.node.rpc_bind,
+        StacksPrivateKey::new(), // We are just reading so don't care what the key is
+        false,
+        next_reward_cycle,
+        SignerSlotID(0), // We are just reading so again, don't care about index.
+    );
+
+    let next_signer_slot_ids: Vec<_> = signer_test
+        .get_signer_indices(next_reward_cycle)
+        .iter()
+        .map(|id| id.0)
+        .collect();
+
+    let mut no_next_signer_messages = || {
+        let _ = wait_for(30, || {
+            for slot_id in next_signer_slot_ids.iter() {
+                let latest_msgs = StackerDB::get_messages::<SignerMessage>(
+                    stackerdb
+                        .get_session_mut(&MessageSlotID::BlockResponse)
+                        .expect("Failed to get BlockResponse stackerdb session"),
+                    &[*slot_id],
```

Comment:
Pretty nit, but you could just do `get_messages(... [next_signer_slot_ids])` instead of iterating through